parser/src/parser.rs

```diff
@@ -152,31 +153,35 @@ impl<'a> Parser<'a> {
             }
 
             _ => self.next_expression_statement(),
-        }
+        };
+
+        self.handle_cast(expr?)
```

Comment:
Only values can be cast. If this change is reverted, the code still passes all the tests.